### DISCUSSION TITLE
Trying to get fade.skip working

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#4192] [General]
I'm trying to make fade.skip working, but failing, and wondering if anyone has made it work, and how.

```
 s = playlist(mode="normal", "./media/test.m3u")

def meta(m)
    [("liq_skip_meta", "false")]
end
s = metadata.map(meta, s)

s = fade.skip(s)

s = insert_metadata(s)

def fade_skip()
    print("Fading out")
    s.insert_metadata([("liq_skip_meta", "true")])
end

thread.run(every=10., fade_skip)

s = source.on_metadata(s, fun(m) -> print(m['liq_skip_meta']))

enc = %mp3(bitrate=32, samplerate=22050, stereo=false)
output.harbor(fallible=true, mount="/test",enc, s)
```
This is what I'm trying, ```liq_skip_meta``` shows as false to start, and then as true when ever the fade_skip function runs.
But no fade and no skip.

I've been testing this on 2.2.4.

Thanks.

**Comment by Thaam** [comment]
Hi,

Put fade.skip() after source.on_metadata().

Tested on 2.2.5 and 2.3.x

**Reply by Russsgithub** [reply]
I think it needs to be after insert_metadata , which makes sense as this is where the metadata is inserted. Thanks @Thaam
